mmeds/log.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import mmeds.config as fig
+import multiprocessing_logging as mpl
+from multiprocessing import current_process
+
+loggers = {}
+mpl.install_mp_handler()
+
+
+class MMEDSLog():
+    global loggers
+
+    def __init__(self, name, testing=False):
+        if loggers.get(name):
+            self.logger = loggers.get(name)
+        else:
+            self.logger = logging.getLogger('Error')
+            formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+            if 'SQL' in name:
+                fh = logging.FileHandler(fig.SQL_LOG)
+            else:
+                fh = logging.FileHandler(fig.MMEDS_LOG)
+            fh.setLevel(logging.DEBUG)
```

Comment:
I might recommend having an option for setting Debug level logging, then any performance impacts from having a lot of logging are safeguarded.

Reply:
I take it you mean having an option to disable debug logging, so only error logs are recorded? I do think it might be worth looking into disabling logs in some situations but I'm not especially worried about the performance impact. With the multiprocess setup the initial calls to the logger are created by the executing process but then handled by a separate process that actually handles the locks and writing them to a file. It may cause some additional disk I/O load but since linux performs writes in memory before flushing them to disk when necessary I'm not super concerned about that either